generators/readme/src/index.ts

```diff
@@ -0,0 +1,170 @@
+import type { VFile, VFileCompatible } from 'vfile';
+import type { Node } from 'unist';
+import type { Parent, Heading, Text, Paragraph, Code, Root, HTML } from 'mdast';
+import type { Options as PrettierOptions } from 'prettier';
+import type {
+  CommandFlags,
+  GeneratorReadmeOptions,
+  PackgeJsonInfo,
+} from './types';
+
+import fs from 'fs';
+import path from 'path';
+import toVfile from 'to-vfile';
+import vfile from 'vfile';
+import unified from 'unified';
+import markdown from 'remark-parse';
+import mdx from 'remark-mdx';
+import stringify from 'remark-stringify';
+import rcfile from 'rcfile';
+import prettier from 'prettier';
+import camelcase from 'lodash/camelCase';
+import capitalize from 'lodash/capitalize';
+import stringfyOptions from './utils/stringify-options';
+
+const prettierConfig = rcfile<PrettierOptions>('prettier');
+
+const html = (value: string): HTML => ({
+  type: 'html',
+  value,
+});
+const text = (value: string): Text => ({
+  type: 'text',
+  value,
+});
+const paragraph = (value: string): Paragraph => ({
+  type: 'paragraph',
+  children: [text(value)],
+});
+const heading = (depth: Heading['depth'], value: string): Heading => ({
+  type: 'heading',
+  depth,
+  children: [text(value)],
+});
+const code = (lang: string, value: string): Code => ({
+  type: 'code',
+  lang,
+  value,
+});
+const parseMarkdownFragmentToAST = (filePath: string) => {
+  const fragmentFile = toVfile.readSync(filePath);
+  const fragmentAST = unified()
+    .use(markdown)
+    .use(stringify, stringfyOptions)
+    .use(mdx)
+    .parse(fragmentFile) as Root;
+
+  return fragmentAST.children;
+};
+
+function readmeTransformer(options: GeneratorReadmeOptions) {
+  const packageJsonRaw = fs.readFileSync(
+    path.join(options.packagePath, 'package.json'),
+    {
+      encoding: 'utf8',
+    }
+  );
+  const parsedPackageJson = JSON.parse(packageJsonRaw);
+  const packageJsonInfo: PackgeJsonInfo = {
+    name: parsedPackageJson.name,
+    description: parsedPackageJson.description,
+    version: parsedPackageJson.version,
+    peerDependencies: parsedPackageJson.peerDependencies,
+  };
+
+  return transformer;
+
+  function transformer(tree: Node, _file: VFile) {
+    (tree as Parent).children = [
+      html(
+        [
+          '<!-- THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. -->',
+          '<!-- This file is created by the `yarn generate-readme` script. -->',
+        ].join('\n')
+      ),
+      // The main heading is derived by the name of the folder
+      heading(1, capitalize(camelcase(path.basename(options.packagePath)))),
+      // The description is what's defined in the package.json
+      heading(2, 'Description'),
+      paragraph(packageJsonInfo.description),
+      // Describe the installation steps
+      heading(2, 'Installation'),
+      code('', `yarn add ${packageJsonInfo.name}`),
+      code('', `npm --save install ${packageJsonInfo.name}`),
```

Comment:
the default code `lang` is empty string. Can we mention `bash` or `sh` if possible?

Reply:
We can but technically it's not really bash.